src/shared/wifi-util.c

```diff
@@ -23,6 +23,13 @@ int wifi_get_interface(sd_netlink *genl, int ifindex, enum nl80211_iftype *iftyp
                 return log_debug_errno(r, "Could not append NL80211_ATTR_IFINDEX attribute: %m");
 
         r = sd_netlink_call(genl, m, 0, &reply);
+        if (r == -ENODEV) {
+                /* For obsolete WEXT driver. */
+                log_debug("Failed to request information about wifi interface %d. "
+                          "The device doesn't seem to have nl80211 interface.",
+                          ifindex);
+                return 0;
```

Comment:
hmm, actually, accordingly to our coding style function should initialize all return params on success, and none on failure. "0" is generally understood to be "success", hence this should initialize the two return parameters here too. (maybe just to 0 and NULL).

(Actually this applies to the pre-existing code, below too)